emscripten.py

```diff
@@ -546,7 +539,7 @@ def finalize_wasm(infile, outfile, memfile, js_syms):
   # EMSCRIPTEN_KEEPALIVE (llvm.used).
   # These are any exports that were not requested on the command line and are
   # not known auto-generated system functions.
-  unexpected_exports = [e for e in metadata.exports if treat_as_user_function(e)]
+  unexpected_exports = [e for e in metadata.all_exports if treat_as_user_function(e)]
```

Comment:
I'm confused as to which places want function export and which all exports. For example, here, isn't "user function" something that only relates to functions, so shouldn't we only look at function exports?

Reply:
The idea behind this change is to try to clear up some of the confusion by using explicit names everywhere.  Hopefully at least now when you read the code you can see which set is being used (even if its possible to still be confused about the why its being used).

I renamed `treat_as_user_function` to `treat_as_user_export` to make it more clear.